sympy/printing/aesaracode.py

```diff
@@ -139,7 +139,7 @@ def _get_or_create(self, s, name=None, dtype=None, broadcastable=None):
         if key in self.cache:
             return self.cache[key]
 
-        value = aet.tensor(name=name, dtype=dtype, broadcastable=broadcastable)
+        value = aet.tensor(name=name, dtype=dtype, shape=broadcastable)
```

Comment:
@brandonwillard is this change safe for both Theano and Aesara?

Reply:
Also another warning that I haven't managed to suppress yet:
```
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/site-packages/aesara/link/c/lazylinker_c.py", line 129, in <module>
    code = open(cfile).read()
ResourceWarning: unclosed file <_io.TextIOWrapper name='/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/site-packages/aesara/link/c/c_code/lazylinker_c.c' mode='r' encoding='UTF-8'>
```
The complaint is about not using `with` but it's coming from the Aesara code.

Reply:
> @brandonwillard is this change safe for both Theano and Aesara?

No, it won't work for Theano.

Reply:
> Also another warning that I haven't managed to suppress yet:
> 
> ```
> Traceback (most recent call last):
>   File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/site-packages/aesara/link/c/lazylinker_c.py", line 129, in <module>
>     code = open(cfile).read()
> ResourceWarning: unclosed file <_io.TextIOWrapper name='/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/site-packages/aesara/link/c/c_code/lazylinker_c.c' mode='r' encoding='UTF-8'>
> ```
> 
> The complaint is about not using `with` but it's coming from the Aesara code.

We can create an Aesara issue for that.

Reply:
> No, it won't work for Theano.

Thanks. I made a spinout PR from this one that included that change but I guess I'll need to revert that and give this more thought: https://github.com/sympy/sympy/pull/23754/files#r917456138

Reply:
> let's see if we can reproduce it in isolation first.

The warning is seen with `-Werror` as in
```console
$ python -Werror myscript.py
```
or alternatively
```console
$ PYTHONWARNINGS=error python myscript.py
```
Either way the line `code = open(cfile).read()` should be changed to use `with` if that line is in the codebase.

Reply:
I've opened a PR to clean up those issues: https://github.com/aesara-devs/aesara/pull/1049.

Reply:
https://github.com/aesara-devs/aesara/releases/tag/rel-2.7.7

Reply:
Thanks.

Another one that I often see is this:
```
>>> import aesara
WARNING (aesara.tensor.blas): Using NumPy C-API based implementation for BLAS functions.
```
That one is a logger warning rather than a Warning warning so it isn't filtered in the usual way. There is some code here to hide it:
https://github.com/sympy/sympy/blob/d98358988cd89fc2c87ec99cbbcce2ba576188fc/sympy/printing/tests/test_aesaracode.py#L18-L21
The problem is that the warning is emitted on import so we would need to guard every place that imports aesara in case some other place imports it first:
https://github.com/sympy/sympy/blob/d98358988cd89fc2c87ec99cbbcce2ba576188fc/sympy/testing/runtests.py#L704

I'm not sure what the best approach is for that warning but emitting a visible message to stderr during import doesn't seem great. Note that the line of code above is just testing to see if aesara is installed but that triggers the warning.